src/utils/unit_test/dnn_lib_test.cpp

```diff
@@ -0,0 +1,365 @@
+////////////////////////////////////////////////////////////////////////////////
+// Copyright (c) 2014-2019, Lawrence Livermore National Security, LLC.
+// Produced at the Lawrence Livermore National Laboratory.
+// Written by the LBANN Research Team (B. Van Essen, et al.) listed in
+// the CONTRIBUTORS file. <lbann-dev@llnl.gov>
+//
+// LLNL-CODE-697807.
+// All rights reserved.
+//
+// This file is part of LBANN: Livermore Big Artificial Neural Network
+// Toolkit. For details, see http://software.llnl.gov/LBANN or
+// https://github.com/LLNL/LBANN.
+//
+// Licensed under the Apache License, Version 2.0 (the "Licensee"); you
+// may not use this file except in compliance with the License.  You may
+// obtain a copy of the License at:
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the license.
+////////////////////////////////////////////////////////////////////////////////
+
+// MUST include this
+#include <catch2/catch.hpp>
+
+#include <lbann/utils/dnn_enums.hpp>
+#include <lbann/utils/dnn_lib/helpers.hpp>
+
+#include <lbann/utils/dnn_lib/convolution.hpp>
+#include <lbann/utils/dnn_lib/dropout.hpp>
+#include <lbann/utils/dnn_lib/local_response_normalization.hpp>
+#include <lbann/utils/dnn_lib/pooling.hpp>
+#include <lbann/utils/dnn_lib/softmax.hpp>
+
+using namespace lbann;
+
+TEMPLATE_TEST_CASE("Tensor operations", "[dnn_lib]", float)
```

Comment:
For other reviewers: In the future, we would expand to include 16-bit FP types, hence `TEMPLATE_TEST_CASE` is a suitable placeholder.